src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/CaseAssignmentControllerFunctionalTest.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.prl.controllers;
+
+import io.restassured.RestAssured;
+import io.restassured.specification.RequestSpecification;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.prl.ResourceLoader;
+import uk.gov.hmcts.reform.prl.utils.IdamTokenGenerator;
+import uk.gov.hmcts.reform.prl.utils.ServiceAuthenticationGenerator;
+
+@Slf4j
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@ContextConfiguration
+public class CaseAssignmentControllerFunctionalTest {
+    @Autowired
+    protected IdamTokenGenerator idamTokenGenerator;
+
+    @Autowired
+    protected ServiceAuthenticationGenerator serviceAuthenticationGenerator;
+
+    private static final String VALID_REQUEST_BODY = "controller/barristerRequest.json";
+
+    private final String targetInstance =
+        StringUtils.defaultIfBlank(
+            System.getenv("TEST_URL"),
+            "http://localhost:4044"
+        );
+
+    private final RequestSpecification request = RestAssured.given().relaxedHTTPSValidation().baseUri(targetInstance);
+
+    @Test
+    public void testAddBarristerAboutToSubmit() throws Exception {
+
+        String requestBody = ResourceLoader.loadJson(VALID_REQUEST_BODY);
+        request
+            .header("Authorization", idamTokenGenerator.generateIdamTokenForSolicitor())
+            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
+            .body(requestBody)
+            .when()
+            .contentType("application/json")
+            .post("/case-assignment/barrister/add/about-to-submit")
+            .then()
+            .assertThat().statusCode(200);
+    }
+
+    @Test
+    public void testRemoveBarristerAboutToSubmit() throws Exception {
+        String requestBody = ResourceLoader.loadJson(VALID_REQUEST_BODY);
+        request
+            .header("Authorization", idamTokenGenerator.generateIdamTokenForSolicitor())
+            .header("ServiceAuthorization", serviceAuthenticationGenerator.generateTokenForCcd())
+            .body(requestBody)
+            .when()
+            .contentType("application/json")
+            .delete("/case-assignment/barrister/remove/about-to-submit")
```

Comment:
I think it should be a post request, all ccd callbacks are post requests

Reply:
Updated to post